string device_name
string device_type
string error_code
string description
string advice
